### ISSUE TITLE
HashSyntax plus HashAlignment breaks the page

### ISSUE THREAD

**Issue by dorner** [rubocop-hq/rubocop#8347]
When mixing `Style/HashSyntax` with a setting of `hash_rockets` and the default `Layout/HashAlignment` cop, auto-correction can break the page and render a syntax error.

## Expected behavior

Original code:
```ruby
some_method(a: 'abc', b: 'abc',
        c: 'abc', d: 'abc'
        )
```

Expected output:
```ruby
some_method(:a => 'abc', :b => 'abc',
            :c => 'abc', :d => 'abc'
        )
  end
```

## Actual behavior

Actual output:

```ruby
some_method(:a => 'abc', :b => 'abc',
        :    c => 'abc', :d => 'abc'
        )
```

## Steps to reproduce the problem

Running `rubocop -a` on the above code will replicate the problem.

## RuboCop version

```
0.88.0 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.7.0 x86_64-darwin16)
```

**Comment by dorner** [comment]
So it looks like the problem is that both cops do inserts - `HashSyntax` inserts the `:` before the key, and `HashAlignment` inserts spaces before the key. It looks like these aren't considered clobbers for some reason.

An easy fix would be to reorder the cops so `HashSyntax` always gets run *after* `HashAlignment`. Probably doesn't address the root cause though.
